contrib/mesos/pkg/executor/executor.go

```diff
@@ -162,24 +145,53 @@ func New(config Config) *Executor {
 		suicideWatch:         &suicideTimer{},
 		shutdownAlert:        config.ShutdownAlert,
 		exitFunc:             config.ExitFunc,
-		podStatusFunc:        config.PodStatusFunc,
 		staticPodsConfigPath: config.StaticPodsConfigPath,
 		launchGracePeriod:    launchGracePeriod,
 		nodeInfos:            config.NodeInfos,
 		initCompleted:        make(chan struct{}),
+		registry:             config.Registry,
+		kubeAPI:              &clientAPIWrapper{config.APIClient},
+		nodeAPI:              &clientAPIWrapper{config.APIClient},
 	}
+
 	runtime.On(k.initCompleted, k.runSendLoop)
 
-	po := newPodObserver(config.PodLW, k.updateTask, k.terminate)
-	runtime.On(k.initCompleted, po.run)
+	k.podWatcher = newPodWatcher(k.registry.watch())
```

Comment:
agreed. i've been thinking about removing broadcaster and ring, just
replacing with a simple buffered chan. i used the broadcaster/ring approach
because I already had the code lying around and it was simple to plug in --
I also thought that I might need multiple watchers on the registry. turns
out that's not the case and so a simple buffered chan is probably the way
to go.

On Fri, Jan 15, 2016 at 10:49 AM, Dr. Stefan Schimanski <
notifications@github.com> wrote:

> In contrib/mesos/pkg/executor/executor.go
> <https://github.com/kubernetes/kubernetes/pull/19632#discussion_r49867768>
> :
>
> >  	runtime.On(k.initCompleted, k.runSendLoop)
> >
> > -	po := newPodObserver(config.PodLW, k.updateTask, k.terminate)
> > -	runtime.On(k.initCompleted, po.run)
> > +	k.podWatcher = newPodWatcher(k.registry.watch())
>
> is this really the only place where a watch is created on the registry? If
> yes, it feels horribly over-engineered to have the broadcaster with its
> slots and the ringbuffer behind that. Maybe I am missing something.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/19632/files#r49867768>.
>